### ISSUE TITLE
Feature parity - `getLegacyAngularInjector().get('timeSrv')`

### ISSUE THREAD

**Issue by tolzhabayev** [grafana/grafana#64400]
### Goal
Check if we need to expose any parts of the `timeSrv` service for angular plugins to be able to upgrade to use React.

### Why?
The `timeSrv` ([implementation](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/services/TimeSrv.ts#L24)) is one of the core Grafana services, and however it was available using the legacy angular injector, is not exposed anymore for the React components. Part of this task is to investigate what plugins are using from this service and if they would need any work from the Grafana core side to be able to migrate.

### Affected plugins
This list is not complete, we were using Github's code search ([cs.github.com](https://cs.github.com/)) to find affected plugins. We are mostly focusing on actively developed plugins / projects here (had contributions in the last 1 year), as they are most likely to be affected by any changes we do to this part of the codebase. 

**Actively developed**
- [grafana/splunk-datasource](https://github.com/grafana/splunk-datasource) (datasource)
    - Usage:
        - Uses `timeSrv.timeRange()` as a fallback value [here](https://github.com/grafana/splunk-datasource/blob/main/src/datasource/query.ts#L29) in case the `request.range` would be undefined in `DataSourceWithBackend.query(request)` ([here](https://github.com/grafana/splunk-datasource/blob/main/src/datasource/Datasource.ts#LL68C8-L68C8)).
    - Migaration: 
        - The `DataQueryRequest.range` property is [always defined](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L499-L500), we could just rely on that.
    
- [grafana-trackmap-panel](https://github.com/pR0Ps/grafana-trackmap-panel) (panel)
    - Usage:
        - Used [here](https://github.com/pR0Ps/grafana-trackmap-panel/blob/master/src/trackmap_ctrl.js#L337) to set the time-range (*this shouldn't be possible from a panel plugin I guess?*)
    - Migration:
        - I don't think we should allow this.
     
- [grafana-pcp](https://github.com/performancecopilot/grafana-pcp) (nested panel plugin)
    - Usage: 
        - calling `timeSrv.setAutoRefresh('')` to turn auto-refresh off in the nested `performancecopilot-flamegraph-panel` panel plugin ([here](https://github.com/performancecopilot/grafana-pcp/blob/main/src/panels/flamegraph/FlameGraphChart.tsx#L38)) 
    - Migration:
        - I don't think we should allow panel plugins to change dashboard-wide settings. They also [knew](https://github.com/performancecopilot/grafana-pcp/commit/4b465ce9a318125672d22008ea1f8b80921e51d9#diff-07816990e431292ee4403e9b532fda0f10f03abfc5e778318858ca59070a613eR44) that this possibility is going to be removed in future versions of Grafana (commit is 2 years old).
     
**Not actively developed**
The following projects seem to depend on the `timeSrv` but are not actively developed (no contributions in the last 1 year): 
- [grafana/plugin-query-engine](https://github.com/grafana/plugin-query-engine)
- [yesoreyeram-boomtable-panel](https://github.com/yesoreyeram/yesoreyeram-boomtable-panel)
- [grafana-3d-globe-panel](https://github.com/satellogic/grafana-3d-globe-panel)
- [monasca-grafana-datasource](https://github.com/openstack/monasca-grafana-datasource)
- [object-datasource-plugin](https://github.com/srclosson/object-datasource)
- [grafana-directed-graph-panel](https://github.com/karmagood/grafana-directed-replication-graph-panel)

### Summary
As far as I could see the affected projects are mostly either:
- try to access the time-range from the query editor, however now this should be accessible through `QueryEditorProps.range` ([link](https://github.com/grafana/grafana/blob/main/public/app/features/query/components/QueryEditorRow.tsx#L285)))
- try to access the time-range from `DataSourceWithBackend.query(request)`, however `request.range` should always be defined

There are plugins which try to access the time-range from the datasource config page (e.g. [object-datasource](https://github.com/srclosson/object-datasource/blob/05d0256476c32b5919b3c1cb495a07870cb61d69/src/Components/QueryLink.tsx#L89)), however I don't feel that it is a well-justified use-case.

**My opinion is that we don't expose `timeSrv` or `timeSrv.timeRange()` for now, at least not until we see a valid use-case.** 

### Action points
- [x] Reach out to @grafana/dashboards-squad on what their opinion is
- [ ] https://github.com/grafana/grafana/issues/45294

**Comment by mckn** [comment]
Note: we should reach out to the dashboard squad to make sure that the API we expose will play nicely with the scenes API.

**Comment by leventebalogh** [comment]
(Unassigned myself until I can actually jump on it.)

**Comment by tolzhabayev** [comment]
Closing as "won't do" - in favour of this being provided by scenes.
